src/components/admin/store/DetailsForm/ItemDetailsForm.tsx

```diff
@@ -0,0 +1,361 @@
+                  onClick={() => {
+                    setMerchPhotos(merchPhotos.filter((_, index) => index !== i));
+                    if (photo.uuid === undefined) {
+                      URL.revokeObjectURL(photo.uploadedPhoto);
+                    }
+                  }}
+                  destructive
+                >
+                  Remove
+                </Button>
+              </li>
+            )}
+          </Draggable>
+          <li>
+            <label className={style.addImage}>
+              <BsPlus aria-hidden />
+              Add image
+              <input
+                type="file"
+                id="cover"
+                accept="image/*"
+                onChange={e => {
+                  const file = e.currentTarget.files?.[0];
+                  e.currentTarget.value = '';
+                  if (file) {
+                    setPhoto(file);
+                  }
+                }}
+              />
+            </label>
+          </li>
+        </ul>
+
+        <ItemOptionsEditor
+          options={options}
+          onOptions={setOptions}
+          optionType={optionType}
+          onOptionType={setOptionType}
+        />
+
+        <DetailsFormItem error={errors.hidden?.message}>
+          <label>
+            <input type="checkbox" {...register('hidden')} />
+            &nbsp;Hide this item from the public storefront
+          </label>
+        </DetailsFormItem>
+
+        <div className={style.submitButtons}>
+          {mode === 'edit' ? (
+            <>
+              <Button type="submit" disabled={loading}>
+                Save changes
+              </Button>
+              <Button onClick={resetForm} disabled={loading} destructive>
+                Discard changes
+              </Button>
+              <Button onClick={deleteItem} disabled={loading} destructive>
+                Delete item
+              </Button>
+            </>
+          ) : (
+            <>
+              <Button type="submit" disabled={loading}>
+                Create item
+              </Button>
+              <Button onClick={resetForm} disabled={loading} destructive>
+                Clear form
+              </Button>
+            </>
+          )}
+        </div>
+      </form>
+      <Cropper
+        file={photo}
+        aspectRatio={1}
+        maxSize={config.file.MAX_MERCH_PHOTO_SIZE_KB * 1024}
+        maxFileHeight={1080}
+        onCrop={async blob => {
+          setMerchPhotos(photos => [...photos, { blob, uploadedPhoto: URL.createObjectURL(blob) }]);
+          setPhoto(null);
+        }}
+        onClose={reason => {
+          setPhoto(null);
+          if (reason === 'cannot-compress') {
+            showToast(
+              'Your image has too much detail and cannot be compressed.',
+              'Try shrinking your image.'
+            );
+          } else if (reason !== null) {
+            showToast('This image format is not supported.');
+          }
+        }}
```

Comment:
It feels like we generally send the same error for every Cropper component, thoughts on bringing this inside the component and not allowing it to be customized?